frontend/public/components/deploy-image.tsx

```diff
@@ -0,0 +1,412 @@
+      .then((imageStreamImport) => {
+        const name = _.get(imageStreamImport, 'spec.images[0].from.name'),
+            status = _.get(imageStreamImport, 'status.images[0].status'),
+            image =_.get(imageStreamImport, 'status.images[0].image'),
+            tag = _.get(imageStreamImport, 'status.images[0].tag');
+        if (status.status === 'Success') {
+          this.setState({
+            loading: false,
+            isi: {
+              name,
+              image,
+              tag,
+              status
+            },
+            name,
+          });
+        } else {
+          this.setState({
+            loading: false,
+            searchError: status.message,
+          });
+        }
+      }, err => this.setState({error: err.message}));
+  };
+
+  save = event => {
+    event.preventDefault();
+
+    this.setState({
+      inProgress: true,
+      error: null
+    });
+
+    const { name, namespace, isi } = this.state;
+
+    const annotations = {
+      'openshift.io/generated-by': 'OpenShiftWebConsole'
+    };
+
+    const volumes = [], volumeMounts = [];
+    let volumeNumber = 0;
+    _.each(isi.image.dockerImageMetadata.Config.Volumes, (value, path) => {
+      volumeNumber++;
+      const volumeName = `${name}-${volumeNumber}`;
+      volumes.push({
+        name: volumeName,
+        emptyDir: {}
+      });
+      volumeMounts.push({
+        name: volumeName,
+        mountPath: path
+      });
+    });
+
+    const ports = parsePorts(isi.image);
+
+    const labels = {app: name};
+
+    const errorState = err => this.setState({error: this.state.error ? `${this.state.error}; ${err.message}` : err.message});
+
+    const promises = [];
+
+    const createResource = (model, obj) => {
+      const promise = k8sCreate(model, obj)
+        .catch(errorState);
+      promises.push(promise);
+    };
+
+    const deploymentConfig = {
+      kind: 'DeploymentConfig',
+      apiVersion: 'apps.openshift.io/v1',
+      metadata: {
+        name,
+        namespace,
+        labels,
+        annotations
+      },
+      spec: {
+        triggers: [{
+          type: 'ConfigChange'
+        }, {
+          type: 'ImageChange',
+          imageChangeParams: {
+            automatic: true,
+            containerNames: [
+              name
+            ],
+            from: {
+              kind: 'ImageStreamTag',
+              name: `${isi.name}:${isi.tag}`,
+              namespace
+            }
+          }
+        }],
+        replicas: 1,
+        selector: labels,
+        template: {
+          metadata: {
+            labels,
+            annotations
+          },
+          spec: {
+            volumes,
+            containers: [{
+              name,
+              image: isi.image.dockerImageMetadata.Config.Image,
+              ports,
+              volumeMounts
+            }],
+            resources: {}
+          }
+        }
+      },
+      status: {}
```

Comment:
Should have mentioned on last review, but the empty `resources` and `status` can be dropped